test/unit/cmap/wire_protocol/responses.test.ts

```diff
@@ -0,0 +1,63 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+
+import { BSON, MongoDBResponse, OnDemandDocument } from '../../../mongodb';
+
+describe('class MongoDBResponse', () => {
+  it('is a subclass of OnDemandDocument', () => {
+    expect(new MongoDBResponse(BSON.serialize({ ok: 1 }))).to.be.instanceOf(OnDemandDocument);
+  });
+
+  context('get isError', () => {
+    it('returns true when ok is 0', () => {
+      const doc = new MongoDBResponse(BSON.serialize({ ok: 0 }));
+      expect(doc.isError).to.be.true;
+    });
+
+    it('short circuits detection of $err, errmsg, code', () => {
+      const doc = new MongoDBResponse(BSON.serialize({ ok: 0 }));
+      expect(doc.isError).to.be.true;
+      expect(doc).to.not.have.property('cache.$err');
+      expect(doc).to.not.have.property('cache.errmsg');
+      expect(doc).to.not.have.property('cache.code');
+    });
+  });
+
+  context('utf8 validation', () => {
+    afterEach(() => sinon.restore());
+
+    context('when enableUtf8Validation is not specified', () => {
```

Comment:
Can we preserve the utf8 validation tests from `commands.test.ts` that actually confirm we throw on invalid utf8 instead of just spying with sinon?

Reply:
My thinking was what coverage does that provide that isn't provided by BSON's testing of the utf validation options/behavior? I'll see if I can add it back the tough part is that hex blob needs to be refactored 😅